src/LibraryUtilities.ts

```diff
@@ -566,48 +285,235 @@ export function updateSections(oldLayoutSpecs: any, newLayoutSpecs: any): void {
+
+    // previousStartingIndex keeps track of the index of typeListNodes when previousInclude starts comparison.
+    let previousStartingIndex = 0;
+
+    // currentStartingIndex keeps track of the index of typeListNodes when current IncludeInfo starts comparison.
+    let currentStartingIndex = 0;
+
+    for (let i = 0; i < includeItemPairs.length; i++) {
+
+        let include = includeItemPairs[i].include;
+        let includeParts = include.path.split(".");
+        let previousIncludeParts = previousIncludePath.split(".");
+
+        // Start from previousStartingIndex if current path is same as or included by previous path.
+        if (i >= 1 && (compareParts(includeParts, previousIncludeParts) >= 0 ||
+            include.path.indexOf(prefix) != -1 &&
+            previousIncludePath.length > 0 &&
+            include.path.startsWith(previousIncludePath))) {
+            t = previousStartingIndex;
+            currentStartingIndex = previousStartingIndex;
+            previousIncludePath = "";
+        }
 
-    // Search for the nodes that are not displayed in library view.
-    let unprocessedNodes: TypeListNode[] = [];
-    let sectionData = convertSectionToItemData(section);
+        // Continue iterating through the remaining includeInfo even if there are no more typeListNodes
+        if (t >= typeListNodes.length) {
+            previousIncludePath = include.path;
+            previousStartingIndex = currentStartingIndex;
+            currentStartingIndex = t;
+            continue;
+        }
+
+        let fullyQualifiedNameParts = typeListNodes[t].fullyQualifiedName.split(".");
+        let node = typeListNodes[t];
+        let parentItem = includeItemPairs[i].parentItem;
+        nodeMatch = false;
+        nodeInculde = false;
 
-    _.each(allNodes, function (node) {
-        if (!node.processed) {
-            console.warn("Item filtered out from library view: " + node.contextData);
-            unprocessedNodes.push(node);
+        if (include.path.indexOf(prefix) == -1) {
+            let compareResult = compareParts(fullyQualifiedNameParts, includeParts);
+
+            if (compareResult == 0) {
+                nodeMatch = true;
+            } else if (compareResult > 0) {
+                nodeInculde = true;
+
+                let newNameParts = -compareResult - 1;
+                if (include.inclusive == false) {
+                    newNameParts = -compareResult;
+                }
+
+                let newName = fullyQualifiedNameParts.slice(newNameParts).join('.');
+
+                buildLibraryItemsFromName(node, parentItem, newName, include.iconUrl);
+            } else if (node.fullyQualifiedName.localeCompare(include.path) < 0) {
+                i--;
+                t++;
+                continue;
+            }
+        } else if (node.fullyQualifiedName.startsWith(include.path)) {
+            nodeInculde = true;
+            let prefixIndex = node.fullyQualifiedName.indexOf(prefix);
```

Comment:
I think prefix handling can be done at the beginning and rest of the code should be same.